P2406-counted-iterator-and-input-iterators.bs

```diff
@@ -262,6 +267,101 @@ performance, can choose using `take` when they know it works correctly for their
 case.
 
 
+# Design alternative
+
+## Motivation
+
+LEWG reviewed R4 of this paper (Issaquah - 2023-02-07 morning) with the
+following poll results:
+
+### Poll outcomes
+
+#### POLL: Reject the NB Comment US-46-107
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 3 <td> 8 <td> 9 <td> 2 <td> 2
+</table>
+
+Outcom: No consensus
+
+WA: Preference to break C++20 sooner
+
+#### POLL: Change `counted_iterator` to have the proposed behavior of `lazy_counted_iterator`, which is a breaking change.
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 6 <td> 12 <td> 2 <td> 1 <td> 3
+</table>
+
+Outcome:
+
+#### POLL: Change iterator category to at most `forward`
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 0 <td> 8 <td> 7 <td> 0 <td> 2
+</table>
+
+Outcome:
+
+#### POLL: Require `forward` as a minimum underlying iterator category
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 0 <td> 3 <td> 5 <td> 3 <td> 3
+</table>
+
+Outcome:
+
+#### POLL: Remove `base` from `lazy_counted_iterator`
+
+<table>
+<tr><th> SF <th> WF <th> N <th> WA <th> SA
+<tr><td> 1 <td> 9 <td> 1 <td> 3 <td> 1
+</table>
+
+Outcome:
+
+
+## Alternative 1: Replace `counted_iterator` with `lazy_counted_iterator`
+
+This alternative propse to change `counted_iterator` to be what this paper
+suggests as `lazy_counted_iterator` or a variation of it.
+
+Issues with this alternative are:
+
+1. This is breaking change (both ABI and source breaking, and depending on the
+   design it might be a silent break of behavior change too)
+2. At least one implementer can't/won't ship it
+3. It would require that we lose `base()` (which LEWG opposed) since otherwise
+   there'd be a behavior-breaking/dangerous change
+
+
+## Alternative 2: Add `lazy_counted_iterator` and deprecate `counted_iterator`
+
+This alternative strives to reduce to additional complexity in teachability and
+direct the users to the right direction.
+
+The proposal here is to adopt `lazy_counted_iterator`, `views::lazy_counted` and
+`lazy_take` as propsed in this paper and mark `counted_iterator`,
+`views::counted` and `take` as deprecated.
+
+
+### Alternative 2 variation II: Partial deprecation
+
+To allow things like `vector | take(n)` to keep being `random_access_iterator`,
+we want to suggest here to deprecate only the cases of `counted_iterator` used
+`input_iterator` or `forward_iterator`, keeping the cases of `output_iterator`
+and `bidirectional_iterator` (and better) as they are.
+
+We also want LWEG to consider the case of `istreambuf_iterator` (or any iterator
+considered as "lazy input iterator", see [[P2578]] for further discussion),
+where current behavior of `take` does the right thing. Forcing the user to use
+`lazy_take` and then incrementing the iterator once again to continue reading
+from the right location seems like another teachability issue.
+
+
```

Comment:
Maybe we should also add a note here that when we talk about deprecation, we could either specify attribute [[deprecated]], or, to allow greater implementer freedom, it could mean adding a non-normative note that the compiler should warn in these cases.